### PR TITLE
Added quantum chemistry quickstart

### PR DESCRIPTION
**Pull request by josh146**
Ported over the quantum chemistry quickstart, and simplified it slightly.

### CODE REVIEW

File: doc/introduction/chemistry.rst [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        'water',
+        geometry,
+        charge=0,
+        multiplicity=1,
+        basis_set='sto-3g',
+        qc_package='pyscf'
+    )
+
+The output variable ``hf_data`` stores the path to the directory containing the file ``'water
+.hdf5'`` with the Hartree-Fock electronic structure of the water molecule.
```
Comment (zeyueN) [inline doc/introduction/chemistry.rst]: are newlines rendered to spaces in the docs?

File: doc/introduction/chemistry.rst [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+find the lowest eigenvalue of the measured Hamiltonian.
+
+First, the qubit Hamiltonian ``h`` has to be converted to a PennyLane
+Hamiltonian using the :func:`~.load_hamiltonian` function:
+
+.. code-block:: python
+
+    hamiltonian = qml.qchem.load_hamiltonian(h)
+
+where ``hamiltonian`` is 2-tuple of coefficients and PennyLane Pauli observables.
```
Comment (zeyueN) [inline doc/introduction/chemistry.rst]: ``hamiltonian`` is a ``pennylane.beta.vqe.Hamiltonian`` object which is represented as a 2-tuple internally. It's perhaps not necessary to be very precise in the quickstart, but I can see it causing confusion from reading this line.

File: doc/introduction/chemistry.rst [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        n_active_electrons=2,
+        n_active_orbitals=2
+    )
+
+where:
+
+* ``h`` is the qubit Hamiltonian of the molecule represented as an
+  `OpenFermion <https://github.com/quantumlib/OpenFermion>`_ ``QubitOperator``, and
+
+* ``nr_qubits`` is the number of qubits  needed to represent it.
```
[the remark this answers is not in the document]
Comment (josh146) [reply doc/introduction/chemistry.rst]:
```suggestion
* ``nr_qubits`` is the number of qubits operators needed to represent it.
```

File: doc/introduction/chemistry.rst (hunk 1 above)
Comment (josh146) [reply doc/introduction/chemistry.rst]: Good catch!
```suggestion
with the Hartree-Fock electronic structure of the water molecule.
```

File: doc/introduction/chemistry.rst (hunk 2 above)
Comment (josh146) [reply doc/introduction/chemistry.rst]: Maybe we should just remove this sentence?

File: doc/introduction/chemistry.rst [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+For more fine-grained control, these functions may be
+called independently as required to obtain the
+electronic Hamiltonian of the molecule represented in the basis of Pauli matrices.
+
+
+Importing molecular structure data
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The atomic structure of a molecule can be imported from an external file using the
+:func:`~.read_structure` function:
```
[the remark this answers is not in the document]
Comment (josh146) [reply doc/introduction/chemistry.rst]: Either is grammatically correct, but :func:~.read_structure function sounds better in my head? But either should work.

File: doc/introduction/chemistry.rst [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    hf_data = qml.qchem.gen_meanfield_data(
+        'water',
+        geometry,
+        charge=0,
+        multiplicity=1,
+        basis_set='sto-3g',
+        qc_package='pyscf'
+    )
+
+The output variable ``hf_data`` stores the path to the directory containing the file ``'water
```
Comment (josh146) [inline doc/introduction/chemistry.rst]:
```suggestion
The output variable ``hf_data`` stores the path to the directory containing the file ``'water.hd5'``
```

File: doc/introduction/chemistry.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (agran2018) [reply doc/introduction/chemistry.rst]: OK

File: doc/introduction/chemistry.rst (hunk 2 above)
Comment (zeyueN) [reply doc/introduction/chemistry.rst]: I think that works!

File: doc/introduction/chemistry.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (zeyueN) [reply doc/introduction/chemistry.rst]: I prefer the former but not strongly. The latter is perhaps slightly easier to follow for someone who's not already familiar with the package.

File: doc/introduction/chemistry.rst (hunk 2 above)
Comment (josh146) [reply doc/introduction/chemistry.rst]: done!

File: doc/introduction/chemistry.rst [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+.. autosummary::
+
+    read_structure
+    gen_meanfield_data
+    gen_active_space
+    gen_hamiltonian_pauli_basis
+
+
+For more fine-grained control, these functions may be
+called independently as required to obtain the
```
Comment (ixfoduap) [inline doc/introduction/chemistry.rst]: I would maybe end the sentence at "... called independently as required."

File: doc/introduction/chemistry.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (ixfoduap) [reply doc/introduction/chemistry.rst]: " the function :func:~.read_structure " is how you would say it in Spanish, not so much in English :)

File: doc/introduction/chemistry.rst [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+    out of the box. Additionally, if `Open Babel <https://openbabel.org/>`_ is
+    installed, `any format recognized by Open Babel
+    <https://openbabel.org/wiki/Category:Formats>`_ is also supported.
+
+    See the :func:`~.read_structure` function for more details.
+
+Solving the Hartree-Fock equations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :func:`~.gen_meanfield_data` function utilizes  `OpenFermion-PySCF <https://github
```
Comment (ixfoduap) [inline doc/introduction/chemistry.rst]:
```suggestion
The :func:`~.gen_meanfield_data` function uses  `OpenFermion-PySCF <https://github
```
Just not a fan of utilizes...

File: doc/introduction/chemistry.rst [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+
+.. _pl_qchem_vqe:
+
+VQE simulations
+---------------
+
+The Variational Quantum Eigensolver (VQE) is a hybrid quantum-classical computational scheme,
+where a quantum computer is used to prepare the trial wave function of a molecule and to measure
+the expectation value of the *electronic Hamiltonian*, while a classical optimizer is used to
+find the lowest eigenvalue of the measured Hamiltonian.
```
Comment (ixfoduap) [inline doc/introduction/chemistry.rst]:
```suggestion
find its ground state.
```

File: doc/introduction/chemistry.rst [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+
+First, the qubit Hamiltonian ``h`` has to be converted to a PennyLane
+Hamiltonian using the :func:`~.load_hamiltonian` function:
+
+.. code-block:: python
+
+    hamiltonian = qml.qchem.load_hamiltonian(h)
+
+We can then use :func:`~.vqe.cost` to automatically create our QNodes
```
Comment (ixfoduap) [inline doc/introduction/chemistry.rst]:
```suggestion
We can then use :func:`~.vqe.cost` to automatically create the QNodes
```

File: doc/introduction/chemistry.rst [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+and define our cost function:
```
Comment (ixfoduap) [inline doc/introduction/chemistry.rst]:
```suggestion
and define the cost function:
```

File: doc/introduction/chemistry.rst (hunk 6 above)
Comment (josh146) [reply doc/introduction/chemistry.rst]:
```suggestion
called independently as required.
```

File: doc/introduction/chemistry.rst [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+
+    read_structure
+    gen_meanfield_data
+    gen_active_space
+    gen_hamiltonian_pauli_basis
+
+
+For more fine-grained control, these functions may be
+called independently as required to obtain the
+electronic Hamiltonian of the molecule represented in the basis of Pauli matrices.
```
Comment (josh146) [inline doc/introduction/chemistry.rst]:
```suggestion
```

**Pull request merged**